app/assets/javascripts/discourse/app/components/topic-timeline.js

```diff
@@ -56,45 +47,47 @@ export default MountWidget.extend(Docking, {
     this.queueRerender();
   },
 
-  dockCheck(info) {
-    const mainOffset = $("#main").offset();
-    const offsetTop = mainOffset ? mainOffset.top : 0;
-    const topicTop = $(".container.posts").offset().top - offsetTop;
-    const topicBottom =
-      $("#topic-bottom").offset().top - $("#main-outlet").offset().top;
+  dockCheck() {
+    const topicTop = domUtils.offset(document.querySelector(".container.posts"))
+      .top;
+    const topicBottom = domUtils.offset(document.querySelector("#topic-bottom"))
```

Comment:
The domUtils says this regarding performance:

https://github.com/discourse/discourse/blob/5ff0b86b5758f39eb7a7692d5f836bf4afd4cd23/app/assets/javascripts/discourse-common/addon/utils/dom-utils.js#L2-L6

I guess it's probably no worse than JQuery's `.offset()` function... but if there's any way we can solve this in another way, that would be great!

Reply:
Yes, good point. I am trying an intersection observer in the last commit, it seems to be working well, but I need to run a few more tests to compare performance. In theory, it should be considerably better.